http/repository/user/repository.go

```diff
@@ -0,0 +1,172 @@
+package userrepository
+
+import (
+	"context"
+	"errors"
+	"strconv"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+
+	"github.com/jumaroar-globant/go-bootcamp/user/pb"
+
+	sharedLib "github.com/jumaroar-globant/go-bootcamp/shared"
+
+	"google.golang.org/grpc"
+)
+
+var (
+	ErrBadAge = errors.New("age is not a number")
+)
+
+type userRepository struct {
+	client pb.UserServiceClient
+	logger log.Logger
+}
+
+// UserRepository is the user repository
+type UserRepository interface {
+	Authenticate(ctx context.Context, username string, password string) (string, error)
+	CreateUser(ctx context.Context, user *sharedLib.User) (*sharedLib.User, error)
```

Comment:
In many cases, you will notice that is better to pass objects through different layers of the application and avoid passing pointers, we can discuss it more in detail.

Reply:
Done